### DISCUSSION TITLE
debugging c++ by skipping the wasm and JS

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#17025] [Q&A]
Hi,
Is any one know how to debug the c++ part (DevTools), but skipping the wasm code ?
it is exhausted to get into the webasm code when we just want to debug only c++.
Skip over (F10) , when you get into the wasm, doesn't get you into the c++ function.

**Comment by sbc100** [comment]
@pfaffe ?

**Comment by pfaffe** [comment]
Can you share any more details? how do you compile your project? a small reproducer would be amazing!

**Comment by aharondavid** [comment]
in the call stack, you can see a c++ and JS entries.
when doing 'debug step in' into  the c++ func, you get into JS function.
now, if you press on 'debug step over', the debugger miss the c++ func and set the debug cursor after the c++ func.
for workaround this, when the debug cursor is in JS, I press 'debug step into' many times, until I get in the c++ func.
  
![image](https://user-images.githubusercontent.com/41375082/170198618-87ca1095-116c-41bc-8fe3-17665740f08e.png)
